app/models/voucher.rb

```diff
@@ -71,6 +71,22 @@ def self.associations_adding_services
     [:unit, :sub_program]
   end
 
+  def available_at
+    cursor = self
+
+    while cursor
+      if cursor.available
```

Comment:
Doesn't this mean if the vacancy isn't currently available this will never catch?  As in, if this vacancy was created, made available, involved in a successful match and then made unavailable, this would return `nil`.  Perhaps, painful though it seems, we need to get all versions with the created_at and available and inspect that array.

Reply:
The `if` is detecting when the availability changes by checking if to see if the version at the cursor is available and the in the previous version it wasn't. If the version at the cursor isn't available, the cursor moves back to the previous version so that all of the history is inspected.

Reply:
Right.  My mistake